src/OmniSharp.DotNetTest/TestFrameworks/TestFramework.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace OmniSharp.DotNetTest.TestFrameworks
+{
+    public abstract class TestFramework
+    {
+        private static readonly ImmutableDictionary<string, TestFramework> s_frameworks;
+
+        static TestFramework()
+        {
+            var builder = ImmutableDictionary.CreateBuilder<string, TestFramework>();
+
+            var nunit = new NUnitTestFramework();
+            var xunit = new XunitTestFramework();
+
+            builder.Add(nunit.Name, nunit);
+            builder.Add(xunit.Name, xunit);
+
+            s_frameworks = builder.ToImmutable();
+        }
+
+        public static TestFramework GetFramework(string name)
+        {
+            return s_frameworks.TryGetValue(name, out var result)
+                ? result
+                : null;
+        }
+
+        public static IEnumerable<TestFramework> GetFrameworks()
+        {
+            foreach (var kvp in s_frameworks)
+            {
+                yield return kvp.Value;
+            }
+        }
+
+        public abstract string FeatureName { get; }
+        public abstract string Name { get; }
+        public abstract string MethodArgument { get; }
+
+        protected abstract bool IsTestAttributeName(string typeName);
```

Comment:
Are there any test frameworks that work with a method "Prefix" / "Postfix" model?

I can't think of one off the top of my head but it's something to consider here in the future, we may need more than just the type name.  Should we also pass in the symbol as well?

ie `public void Test_ShouldDoSomethingCool()` or `public void RunAndDoesTest()`?
Or perhaps something like XUnit's skip `[Fact(Skip="")]` we could technically not consider that as a test, because it won't do anything.